### PR TITLE
Add mainModuleId and mainFunction to manifest

### PR DESCRIPTION
**Pull request by adrw**
For easier review and more iterative development, I spiked this initial change out of a PR that kept growing and will land it as follow ups instead of blocking this manifest API change further. https://github.com/cashapp/zipline/pull/560

### CODE REVIEW

File: zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt [hunk 1]
```diff
@@ -45,20 +45,20 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
     version = BuildConfig.KOTLIN_PLUGIN_VERSION,
   )
 
-  override fun apply(project: Project) {
-    super.apply(project)
+  override fun apply(target: Project) {
```
Comment (adrw) [inline zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt:48]: Kotlin compiler warning that the parameter name didn't match the supertype's.

File: zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineCompileTask.kt [hunk 2]
```diff
@@ -48,6 +50,12 @@ abstract class ZiplineCompileTask @Inject constructor(
   @get:OutputDirectory
   val outputDir: DirectoryProperty = objectFactory.directoryProperty()
 
+  @get:Input
+  val mainModuleId: Property<String> = objectFactory.property(String::class.java)
```
Comment (adrw) [inline zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineCompileTask.kt:54]: For now, we'll pass in the `mainModuleId` and `mainFunction` through the plugin.

File: samples/trivia/trivia-js/build.gradle.kts [hunk 3]
```diff
@@ -1,6 +1,4 @@
 import org.jetbrains.kotlin.gradle.plugin.PLUGIN_CLASSPATH_CONFIGURATION_NAME
-import org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootExtension
```
Comment (adrw) [inline samples/trivia/trivia-js/build.gradle.kts:2]: Unused imports

File: zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineCompileTask.kt (hunk 2 above)
Comment (swankjesse) [reply zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineCompileTask.kt:54]: Love it

**Review by swankjesse — APPROVED** [review]

**Comment by swankjesse** [comment]
Wonderful!

**Pull request merged**